app/models/miq_queue.rb

```diff
@@ -545,6 +545,10 @@ def check_for_timeout(log_prefix = "MIQ(MiqQueue.check_for_timeout)", grace = 10
     end
   end
 
+  def self.check_for_timeout
+    where(:state => STATE_DEQUEUE).where("(select date_part('epoch', updated_on) + msg_timeout) < ?", Time.now.to_i).each(&:check_for_timeout)
```

Comment:
Minor, but can we move out the query into it's own scope method?  Maybe `candidates_for_timeout`?